src/Proxy.hs

```diff
@@ -1341,3 +1356,35 @@ getPassiveDelegationR =
   where
     doGetPassiveDelegation = withLastFinalBlockHash Nothing (getPoolStatus (BakerId $ AccountIndex 0) True)
 
+matchesVersion :: Word -> Maybe ForcedUpdateConfig -> AE.Value
+matchesVersion _ Nothing = AE.object ["status" AE..= String "ok"]
+matchesVersion queryVersion (Just ForcedUpdateConfig{..})
+  | inRanges fucForceUpdate queryVersion = AE.object [
+      "status" AE..= String "needsUpdate",
+      "url" AE..= fucURL
+    ]
+  | inRanges fucSuggestUpdate queryVersion = AE.object [
+      "status" AE..= String "warning",
+      "url" AE..= fucURL
+      ]
+  | otherwise = AE.object ["status" AE..= String "ok"]
+
+getAppSettings :: Handler TypedContent
+getAppSettings = do
+  Proxy{..} <- getYesod
+  mPlatform <- lookupGetParam "platform"
+  mVersion <- lookupGetParam "appVersion"
+  case (Text.strip <$> mPlatform, readMaybe . Text.unpack =<< mVersion) of
```

Comment:
Normally I'm in favor of stripping/trimming user input, but wouldn't in this case as it should be all programmatic.

Reply:
Removed stripping of the platform.